tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/RabbitMQ/ExchangeDeclareIntegration.cs

```diff
@@ -43,8 +43,9 @@ public class ExchangeDeclareIntegration
         /// <param name="arguments">The original arguments setting</param>
         /// <returns>Calltarget state value</returns>
         internal static CallTargetState OnMethodBegin<TTarget>(TTarget instance, string exchange, string type, bool passive, bool durable, bool autoDelete, bool @internal, bool nowait, IDictionary<string, object> arguments)
+            where TTarget : IModelBase
         {
-            return new CallTargetState(RabbitMQIntegration.CreateScope(Tracer.Instance, out _, Command, SpanKinds.Client, exchange: exchange));
+            return new CallTargetState(RabbitMQIntegration.CreateScope(Tracer.Instance, out _, Command, SpanKinds.Client, exchange: exchange, host: instance.Session.Connection.Endpoint.HostName));
```

Comment:
Do we need a defensive null check here?

```suggestion
            return new CallTargetState(RabbitMQIntegration.CreateScope(Tracer.Instance, out _, Command, SpanKinds.Client, exchange: exchange, host: instance?.Session.Connection.Endpoint.HostName));
```